server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3973,50 +3981,49 @@ private Pair<List<Long>, Integer> searchForServiceOfferingIdsAndCount(ListServic
         }
 
         if (vmTypeStr != null) {
-            sc.setParameters("vmType", vmTypeStr);
+            sc.setParameters("svmType", vmTypeStr);
         }
 
         useStorageType(sc, storageType);
 
         if (zoneId != null) {
-            sc.setJoinParameters("ZoneDetailSearch", "name", "zoneid");
-            sc.setJoinParameters("ZoneDetailSearch", "value", zoneId);
+            sc.setJoinParameters("ZoneDetailSearch", "zoneId", zoneId);
 
             if (DataCenter.Type.Edge.equals(zone.getType())) {
                 sc.setJoinParameters("diskOfferingSearch", "useLocalStorage", true);
             }
         }
 
         if (cpuNumber != null) {
-            sc.setJoinParameters("maxComputeDetailsSearch", "name", "maxcpunumber");
-            sc.setJoinParameters("minComputeDetailsSearch", "name", "mincpunumber");
-            sc.setParameters("cpuConstraintEQ", cpuNumber);
-            sc.setParameters("cpuConstraintMinComputeLTEQ", cpuNumber);
-            sc.setParameters("cpuConstraintMaxComputeGTEQ", cpuNumber);
+            /*
+            and("cpuConstraintMinComputeGTEQ", entity().getCpu(), Op.GTEQ).cp();
+            or("cpuConstraintGTEQ", entity().getCpu(), Op.GTEQ);
+
+            or().op().op();
+
+            and("cpuConstraintMinComputeLTEQ", entity().getCpu(), Op.LTEQ).cp();
+            or("minComputeDetailsSearch", "cpuConstraintMinComputeLTEQ", minComputeDetailsSearch.entity().getValue(), Op.LTEQ).cp();
+
+            and("cpuConstraintMaxComputeGTEQ", entity().getCpu(), Op.GTEQ).cp();
+            or("maxComputeDetailsSearch", "cpuConstraintMaxComputeGTEQ", maxComputeDetailsSearch.entity().getValue(), Op.GTEQ).cp();
+            cp().cp();
+             */
```

Comment:
comment in code. that said, is it maybe time to disect the query manager and have it constructed of the manager and some join/view objects. It has become rather big and is shrinking in maintainability.